### PR TITLE
Refactor install-strategy package 

### PR DESCRIPTION
**Pull request by ashleyschuett**
**What this PR does / why we need it**:

Organizing the reconcile packages for kubevirt resources. This will make the forth coming "strict-reconcile" PRs easier to review. I have also broken up the tests that where under "create" to be in individual test files for the component that is being tested. 

Current: 
```
pkg/virt-operator
└───creation
│   └───components
│   └───csv
│   └───rbac
└───install-strategy
```

Proposed: 
```
pkg/virt-operator
└───resources
│   └───creation
│   │   └───components
│   │   └───csv
│   │   └───rbac
│   │   └───install
│   └───apply

```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by davidvossel — COMMENTED** [review]
/approve

changes look great. I've questioned whether the existing `creation` directory was accurately named. I've thought that `generate` might be more accurate since we're generating objects. Creation makes it sound like we're publishing them to the api server or something, which is not the case. 

I don't feel that strongly either way. If you decided to make the change, it's probably as simple as running this sed command at the root of the source tree. 

```
find . -type f -exec sed -i "s|virt-operator/resource/creation|virt-operator/resource/generate|g" {} \;

```

**Comment by ashleyschuett** [comment]
@davidvossel thank you for the review! I think you are correct and have updated the PR with your suggestion.

File: pkg/virt-operator/resource/generate/install/strategy.go [hunk 1]
```diff
@@ -292,42 +368,28 @@ func GenerateCurrentInstallStrategy(config *operatorutil.KubeVirtDeploymentConfi
...
-		productName = config.GetProductName()
-	} else {
-		log.Log.Errorf("invalid kubevirt.spec.productName: labels must be 63 characters or less, begin and end with alphanumeric characters, and contain only dot, hyphen or dash")
-	}
-	if isValidLabel(config.GetProductVersion()) {
-		productVersion = config.GetProductVersion()
-	} else {
-		log.Log.Errorf("invalid kubevirt.spec.productVersion: labels must be 63 characters or less, begin and end with alphanumeric characters, and contain only dot, hyphen or dash")
-	}
-
```
Comment (davidvossel) [inline pkg/virt-operator/resource/generate/install/strategy.go:308]: there's a lot of indirection going on here, but ultimately this corresponds to the kv.Spec.ProductVersion field on the kubevirt cr. I think we still need this. Maybe this was simply moved and I'm missing it?

Comment (ashleyschuett) [reply pkg/virt-operator/resource/generate/install/strategy.go:308]: From my understanding this code was redundant, . We are [injecting these values on reconcile](https://github.com/kubevirt/kubevirt/blob/ff2218cb391acafae35a094c0303ebc0536da490/pkg/virt-operator/resource/apply/reconcile.go#L87-L110). Having them written in the configmap seems confusing. If the operator removes them from the KV config they will then persist, but only for deployments and daemonsets. However, on second inspection it looks like product version is always filled for these resources with the KubeVirt version if the product version is not set. So I will add this back. But it does seem a bit confusing and disjointed.

Comment (davidvossel) [reply pkg/virt-operator/resource/generate/install/strategy.go:308]: Ohhhh, i get it. yep I agree.

**Review by davidvossel — APPROVED** [review]
/lgtm

**Review by davidvossel — APPROVED** [review]
/lgmt

**Review by davidvossel — APPROVED** [review]
/lgtm

**Pull request merged**